view/frontend/web/js/paypal_express/button.js

```diff
@@ -0,0 +1,654 @@
+    'Adyen_ExpressCheckout/js/helpers/redirectToSuccess',
+    'Adyen_ExpressCheckout/js/helpers/setExpressMethods',
+    'Adyen_ExpressCheckout/js/helpers/validatePdpForm',
+    'Adyen_ExpressCheckout/js/model/config',
+    'Adyen_ExpressCheckout/js/model/countries',
+    'Adyen_ExpressCheckout/js/model/totals',
+    'Adyen_ExpressCheckout/js/model/currency',
+    'knockout',
+    'Magento_Customer/js/model/customer',
+    'Magento_Checkout/js/model/quote',
+    'mage/url',
+    'Adyen_ExpressCheckout/js/actions/initPayments',
+    'Adyen_ExpressCheckout/js/actions/updatePaypalOrder',
+    'Adyen_ExpressCheckout/js/actions/setBillingAddress',
+    'Magento_Checkout/js/model/full-screen-loader',
+    'Adyen_ExpressCheckout/js/model/adyen-payment-service',
+    'Adyen_ExpressCheckout/js/helpers/getApiUrl',
+    'jquery',
+    'Adyen_ExpressCheckout/js/model/virtualQuote',
+    'Adyen_ExpressCheckout/js/model/maskedId',
+], function (
+    Component,
+    $t,
+    customerData,
+    AdyenConfiguration,
+    AdyenCheckout,
+    activateCart,
+    cancelCart,
+    createPayment,
+    createOrder,
+    getShippingMethods,
+    getExpressMethods,
+    setShippingInformation,
+    setTotalsInfo,
+    formatAmount,
+    getPaypalStyles,
+    getCartSubtotal,
+    getExtensionAttributes,
+    getPaymentMethod,
+    getPdpForm,
+    getPdpPriceBox,
+    getSupportedNetworks,
+    isConfigSet,
+    getRegionId,
+    redirectToSuccess,
+    setExpressMethods,
+    validatePdpForm,
+    configModel,
+    countriesModel,
+    totalsModel,
+    currencyModel,
+    ko,
+    customer,
+    quote,
+    urlBuilder,
+    initPayments,
+    updatePaypalOrder,
+    setBillingAddress,
+    fullScreenLoader,
+    adyenPaymentService,
+    getApiUrl,
+    $,
+    virtualQuoteModel,
+    maskedIdModel
+) {
+    'use strict';
+
+    return Component.extend({
+        isPlaceOrderActionAllowed: ko.observable(
+            quote.billingAddress() != null),
+
+        defaults: {
+            shippingMethods: {},
+            isProductView: false,
+            maskedId: null,
+            paypalComponent: null,
+            shippingAddress: {},
+            shippingMethod: null,
+            shopperEmail: null,
+            billingAddress : {},
+            orderId: null,
+            quoteId: null
+        },
+
+        initialize: async function (config, element) {
+            this._super();
+
+            // Set the config and countries model
+            configModel().setConfig(config);
+            countriesModel();
+
+            // Determine if this is a product view page
+            this.isProductView = config.isProductView;
+
+            if (!this.isProductView) {
+                // Retrieve the PayPal payment method
+                let paypalPaymentMethod = await getPaymentMethod('paypal', this.isProductView);
+                virtualQuoteModel().setIsVirtual(false);
+                paypalPaymentMethod.configuration.intent = 'authorize';
```

Comment:
Wouldn't this throw an error if the method is not available as it appears that we're checking in line 125?